sqlmesh/core/context.py

```diff
@@ -452,18 +427,53 @@ def local_snapshots(self) -> t.Dict[str, Snapshot]:
         """Generates and returns snapshots based on models registered in this context without reconciling them
         with the persisted state.
         """
+        prod = self.state_reader.get_environment(c.PROD)
+        remote_snapshots = (
+            {
+                snapshot.name: snapshot
+                for snapshot in self.state_reader.get_snapshots(prod.snapshots).values()
```

Comment:
The reason I introduced `local_snapshots` is to not require connection to the state sync (eg. when running unit tests) as part of https://github.com/TobikoData/sqlmesh/pull/507. Any chance we can somehow move this logic into `def snapshots` and have an optional `external_snapshots` argument or something?